### PR TITLE
rocksdb: 5.11.3 -> 6.1.2

### PR DESCRIPTION
**Pull request by magenbluten**
###### Motivation for this change

rocksdb was out of date. also removed stuff that seemed broken.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by ryantm** [comment]
cc @adevress

File: pkgs/development/libraries/rocksdb/default.nix
Comment (ryantm) [inline pkgs/development/libraries/rocksdb/default.nix]:
```suggestion
  pname = "rocksdb";
```

File: pkgs/development/libraries/rocksdb/default.nix
Comment (ryantm) [inline pkgs/development/libraries/rocksdb/default.nix]:
```suggestion
    repo = pname;
```

File: pkgs/development/libraries/rocksdb/default.nix
Comment (ryantm) [inline pkgs/development/libraries/rocksdb/default.nix]: Is `checkInputs` needed? Does cmake automatically do some kind of checking? If we need to turn of checking, can we use `doCheck = false;` instead?

File: pkgs/development/libraries/rocksdb/default.nix
Comment (ryantm) [inline pkgs/development/libraries/rocksdb/default.nix]: cmake automatically enables parallel building
```suggestion
```

**Review by ryantm — CHANGES_REQUESTED** [review]

**Comment by ryantm** [comment]
@GrahamcOfBorg build rocksdb

File: lib/licenses.nix
Comment (ryantm) [inline lib/licenses.nix]: This license is already in the file as `asl20`.

**Review by ryantm — CHANGES_REQUESTED** [review]

File: lib/licenses.nix
Comment (magenbluten) [reply lib/licenses.nix]: sorry. missed that.

Comment (Mic92) [inline lib/licenses.nix]: Please also remove this.

**Review by Mic92 — APPROVED** [review]
Looks good to me otherwise.

File: pkgs/development/libraries/rocksdb/default.nix
Comment (Mic92) [inline pkgs/development/libraries/rocksdb/default.nix]: Could this break somewhere?

File: pkgs/tools/system/osquery/default.nix
Comment (Ma27) [inline pkgs/tools/system/osquery/default.nix]: What exactly broke `osquery`'s build here? And have you checked if there are some upstream patches to work around the problem?

Comment (magenbluten) [reply pkgs/tools/system/osquery/default.nix]: actually, the osquery nixpkg is quite outdated. a plugin of osquery ( in the current source ) depends on rocksdb-lite. the plugin expects the lite version of rocksdb to be linkable via -lrocksdb_lite. but since i didn't have time and i didn't understand the osquery nixpkg ( looks quite complicated ). hence, i decided to mark osquery broken. i need to be repackaged anyways.

Comment (Mic92) [reply pkgs/tools/system/osquery/default.nix]: @flokli maybe regarding osquery?

Comment (Ma27) [reply pkgs/tools/system/osquery/default.nix]: In fact I'm on of the maintainers of osquery, that's the original reason why I asked.
In the end @magenbluten is right, osquery is behind upstream by at least one minor. I tried to do the bump last month (hoping that this would solve several issues), but I faced some hard-to-solve erors back then. I hope that I have sufficient time soon to do the bump.

File: pkgs/top-level/all-packages.nix
Comment (Lassulus) [inline pkgs/top-level/all-packages.nix]: I guess you can remove this also, since this isn't supported anymore

File: pkgs/tools/system/osquery/default.nix
Comment (magenbluten) [reply pkgs/tools/system/osquery/default.nix]: @Ma27 if you could disable the rocksdb plugin or `fix` the CMakefile to link against `-lrocksdb` and not `-lrocksdb_lite` everything should work as expected. However, the rocksdb plugin says it wants rocksdb 5.7.2 afair, this is also outdated. the current release of osquery is 3.3.2 whereas the nixpkg version is 3.2.9.

Comment (Ma27) [reply pkgs/tools/system/osquery/default.nix]: Hmm I see. Admittedly I'm slightly hesitant to patch the CMake files even further as it diverges pretty much from upstream already.

Due to `osquery` being slightly outdated and causing issues with further packages I guess that it should be fine for now to mark it as broken. I already have a local branch where I tried to bump/repackage osquery, but it's far from done. I hopefully have sufficient time for some more hacking next week, I can give this a try then :)

**Comment by Lassulus** [comment]
@GrahamcOfBorg build ceph

**Comment by Lassulus** [comment]
this also breaks ceph. and everything that depends on it. But ceph seems to be an old version? @adevress  @alexanderkjeldaas

**Comment by Lassulus** [comment]
since no one responded for the ceph breakage, we marked ceph as broken.

File: pkgs/development/libraries/rocksdb/default.nix
Comment (flokli) [inline pkgs/development/libraries/rocksdb/default.nix]: already set by `pkgs/development/tools/build-managers/cmake/setup-hook.sh`.

File: pkgs/development/libraries/rocksdb/default.nix
Comment (flokli) [inline pkgs/development/libraries/rocksdb/default.nix]: why was this not enabled?

**Review by flokli — COMMENTED** [review]
some nitpicks.

File: pkgs/development/libraries/rocksdb/default.nix
Comment (magenbluten) [reply pkgs/development/libraries/rocksdb/default.nix]: for some reason cmake could find zlib. didn't know how to fix it due to lack of time. but should be figured out why at some point. did work before v6.

**Comment by magenbluten** [comment]
i tried to unbreak osquery with no luck. i failes when linking against rocksdb(-lite). see https://gist.github.com/magenbluten/997142eaa828ccd177c0dd06cc8261b4

**Comment by Lassulus** [comment]
@Ma27 any points against merging this? since you recently updated osquery and this PR would break it.

**Comment by Ma27** [comment]
No. While working on `osquery` quite recently I realized that they currently have a hard time staying up-to-date with new software versions and I don't consider `osquery` important enough to actually block this PR any longer.

Therefore, it's IMHO ok to merge this now, I'll take care of `osquery` on one of the next weekends :)

**Comment by Ma27** [comment]
@magenbluten @Lassulus I managed to get `osquery` building locally (didn't test `ceph` as a bunch of python2 deps appear to be broken on master atm) and it works perfectly fine when using the old build system (no cmake). I suspect this is because the cmake build doesn't create statically linked libraries.

According to the [install section](https://github.com/facebook/rocksdb/blob/master/INSTALL.md) it's recommended to simply use `make {shared,static}_lib` depending on what's needed.

I'd simply go back to the old build approach for rocksdb (and keep 6.1.2), however I'm not sure if I'm missing a reason why cmake is used now (and I probably just didn't manage to get the static lib building with cmake ^^).

Do you think using the old make-based approach makes sense or am I missing something? :)

**Comment by magenbluten** [comment]
i guess the install section you mention is outdated (dez. 2018). 

the "correct" solution would be to figure out how to make cmake make static libs ;)

**Comment by magenbluten** [comment]
git grep -i static_lib:

CMakeLists.txt:set(ROCKSDB_STATIC_LIB rocksdb${ARTIFACT_SUFFIX})

**Pull request merged**
